src/eppo-client.ts

```diff
@@ -33,23 +35,34 @@ export interface IEppoClient {
 
 export default class EppoClient implements IEppoClient {
   constructor(
-    private configurationRequestor: ExperimentConfigurationRequestor,
+    private configurationStore: EppoLocalStorage,
+    private sessionStorage: EppoSessionStorage,
     private assignmentLogger?: IAssignmentLogger,
   ) {}
 
   getAssignment(subjectKey: string, experimentKey: string, subjectAttributes = {}): string {
     validateNotBlank(subjectKey, 'Invalid argument: subjectKey cannot be blank');
     validateNotBlank(experimentKey, 'Invalid argument: experimentKey cannot be blank');
-    const experimentConfig = this.configurationRequestor.getConfiguration(experimentKey);
+    // If getAssignment was called when the latest configuration were still being downloaded
+    // use the old assignment for the remainder of the session to avoid a flickering effect;
+    // we don't want a subject to switch between 2 variations during the same session.
+    const sessionOverrideKey = `${subjectKey}-${experimentKey}-session-override`;
+    const sessionOverride = this.sessionStorage.get(sessionOverrideKey);
+    if (sessionOverride) {
+      return sessionOverride === NULL_SENTINEL ? null : sessionOverride;
```

Comment:
I didn't invoke the custom logger here, since the assignment will have been logged already during the session (in the scenario where we use an old RAC).